### PR TITLE
Move unbind to the Bound to simplify usage

### PR DESCRIPTION
**Pull request by bogdandrutu**
I think this simplifies the usages of the bind/unbind pair.

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java [hunk 1]
```diff
@@ -20,9 +20,9 @@
  * Base interface for all metrics with bounds defined in this package.
  *
  * @param <B> the specific type of Bound Instrument this instrument can provide.
- * @since 0.1.0
+ * @since 0.3.0
  */
-public interface InstrumentWithBinding<B> extends Instrument {
+public interface InstrumentWithBind<B extends InstrumentWithBind.Bound> extends Instrument {
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: I think we should stick with the existing name.

File: api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java [hunk 2]
```diff
@@ -37,12 +37,15 @@
...
 
-  /**
-   * Removes the {@code Bound Instrument} from the Instrument. i.e. references to previous {@code
-   * Bound Instrument} are invalid (not being managed by the instrument).
-   *
-   * @param boundInstrument the {@code Bound Instrument} to be removed.
-   * @since 0.1.0
-   */
-  void unbind(B boundInstrument);
+  interface Bound {
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: I think this should be called "Binding"

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: I am confused about this strange language called English :))

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: 🤣  yeah, it's a pretty crazy language

File: api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: Done.

File: api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java (hunk 2 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: Went with `BoundInstrument`, does that work?

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/InstrumentWithBind.java]: love it. ❤️

**Pull request merged**